src/puppetlabs/puppetdb/anonymizer.clj

```diff
@@ -31,11 +32,10 @@
    (contains? event "old_value")
    (contains? event "resource_type")))
 
-(defn report?
+(pls/defn-validated report? :- report/report-schema
```

Comment:
Maybe we need to remove the anonymisation out of the equation, if we require more keywordize work to make this work consistenly, perhaps thats for another day. Right now the fact? report? crud, is using manual validation, maybe just make report? work that way, its consistently terrible, but at least it avoids the need to fix the keywordize stuff straight away.